packages/neon_framework/lib/src/blocs/weather_status.dart

```diff
@@ -50,6 +50,12 @@ class _WeatherStatusBloc extends InteractiveBloc implements WeatherStatusBloc {
     });
 
     timer = TimerBloc().registerTimer(const Duration(minutes: 5), refresh);
+
+    location.listen((locationResult) async {
+      if (!locationResult.isLoading && !locationResult.hasError) {
+        await refreshForecast();
```

Comment:
This is missing the condition to check if the location is actually set.
If it is set then the method needs to be executed, otherwise a `Result.success(null)` needs to be added to the subject to clear the old data (the type needs be nullable).